### PR TITLE
Make use of php://memory stream for Unit testing

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #42

### CODE REVIEW

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/ClassCache.php [hunk 1]
```diff
@@ -37,7 +38,8 @@ public function __construct(
      */
     public function set(string $className, string $baseProxyClassName, string $classContents): void
     {
-        file_put_contents($this->getClassPath($className, $baseProxyClassName), "<?php\n\n" . $classContents, LOCK_EX);
+        $flags = $this->cachePath === 'php://memory' ? LOCK_EX : 0;
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/ClassCache.php]: Ooops,,, Yes. But at the end it works fine with exclusive lock so I reverted these changes completely.

File: src/ClassCache.php [hunk 2]
```diff
@@ -82,6 +83,10 @@ public function getClassPath(string $className, string $baseProxyClassName): str
     {
         [$classFileName, $classFilePath] = $this->getClassFileNameAndPath($className, $baseProxyClassName);
 
+        if ($this->cachePath === 'php://memory') {
```
Comment (arogachev) [inline src/ClassCache.php:86]: Just noticed that every time memory stream is used, a bunch of proxy files is left after running tests (location is repo root folder). :thinking:

File: src/ClassCache.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (arogachev) [reply src/ClassCache.php]: Nope, there was another error. Reverted and fixed.

**Comment by arogachev** [comment]
https://github.com/yiisoft/proxy/issues/42#issuecomment-1177092160

**Pull request closed without merging**